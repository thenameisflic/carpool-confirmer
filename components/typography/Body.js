import React from 'react';
import { Text, StyleSheet } from "react-native";

export default function Body({children, style}) {
    return <Text style={[styles.body, style]}>{children}</Text>;
}

const styles = StyleSheet.create({
    body: {
        fontSize: 16,
        fontFamily: "Nunito Sans Regular"
    }
});
