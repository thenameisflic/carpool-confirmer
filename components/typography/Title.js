import React from "react";
import { Text, StyleSheet } from "react-native";

export default function Title({children, style, size = "h1"}) {
    return <Text style={[styles[size], style]}>{children}</Text>
}

const styles = StyleSheet.create({
    h1: {
        fontSize: 96,
        fontFamily: "Nunito Light"
    },
    h2: {
        fontSize: 60,
        fontFamily: "Nunito Light"
    },
    h3: {
        fontSize: 48,
        fontFamily: "Nunito Regular"
    },
    h4: {
        fontSize: 34,
        fontFamily: "Nunito Regular"
    },
    h5: {
        fontSize: 24,
        fontFamily: "Nunito Regular"
    },
    h6: {
        fontSize: 20,
        fontFamily: "Nunito Bold"
    }
});
