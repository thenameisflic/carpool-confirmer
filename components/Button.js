import React from "react";
import { TouchableOpacity, StyleSheet } from "react-native";
import theme from "../theme";
import Title from "./typography/Title";

export default function Button({ onPress, title, style }) {
  return (
    <TouchableOpacity onPress={onPress} style={[styles.container, style]}>
      <Title size="h6" style={styles.text}>{title}</Title>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.primaryColor,
    alignItems: "center",
    justifyContent: "center",
    height: 48,
    borderRadius: 4
  },
  text: {
      color: theme.backgroundColor
  }
});
