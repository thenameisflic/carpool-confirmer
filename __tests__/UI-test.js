import "react-native";
import React from "react";
import Button from "../components/ui/Button";
import Title from "../components/typography/Title";
import Body from "../components/typography/Body";

import renderer from "react-test-renderer";

it("renders Buttons correctly", () => {
  const title = "Test Button";
  const tree = renderer.create(<Button title={title} />).toJSON();
  expect(tree.children[0].children[0]).toEqual(title);
  expect(tree).toMatchSnapshot();
});

it("renders Titles correctly", () => {
  const title = "Test Title";
  const tree = renderer.create(<Title>{title}</Title>).toJSON();
  expect(tree.children[0]).toEqual(title);
  expect(tree).toMatchSnapshot();
});

it("renders Body correctly", () => {
  const title = "Test Body";
  const tree = renderer.create(<Body>{title}</Body>).toJSON();
  expect(tree.children[0]).toEqual(title);
  expect(tree).toMatchSnapshot();
});
