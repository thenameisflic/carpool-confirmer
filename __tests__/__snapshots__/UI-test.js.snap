// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Body correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Nunito Sans Regular",
        "fontSize": 16,
      },
      undefined,
    ]
  }
>
  Test Body
</Text>
`;

exports[`renders Buttons correctly 1`] = `
<View
  accessible={true}
  clickable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FF5E57",
      "borderRadius": 4,
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Nunito Bold",
          "fontSize": 20,
        },
        Object {
          "color": "#fff",
        },
      ]
    }
  >
    Test Button
  </Text>
</View>
`;

exports[`renders Titles correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Nunito Light",
        "fontSize": 96,
      },
      undefined,
    ]
  }
>
  Test Title
</Text>
`;
